pkg/reader/ccip_test.go

```diff
@@ -371,3 +371,288 @@ func TestCCIPChainReader_Sync_BindError(t *testing.T) {
 	require.Error(t, err)
 	require.ErrorIs(t, err, expectedErr)
 }
+
+func TestCCIPChainReader_DiscoverContracts_HappyPath(t *testing.T) {
+	ctx := tests.Context(t)
+	destChain := cciptypes.ChainSelector(1)
+	sourceChain1 := cciptypes.ChainSelector(2)
+	sourceChain2 := cciptypes.ChainSelector(3)
+	s1Onramp := []byte{0x1}
+	s2Onramp := []byte{0x2}
+	destNonceMgr := []byte{0x3}
+	destExtended := reader_mocks.NewMockExtended(t)
+
+	// mock the call for sourceChain1
+	destExtended.EXPECT().ExtendedGetLatestValue(
+		mock.Anything,
+		consts.ContractNameOffRamp,
+		consts.MethodNameGetSourceChainConfig,
+		primitives.Unconfirmed,
+		map[string]any{"sourceChainSelector": sourceChain1},
+		mock.Anything,
+	).Return(nil).Run(func(
+		ctx context.Context,
+		contractName,
+		methodName string,
+		confidenceLevel primitives.ConfidenceLevel,
+		params,
+		returnVal interface{}) {
+		v := returnVal.(*sourceChainConfig)
+		v.OnRamp = s1Onramp
+		v.IsEnabled = true
+	})
+	// mock the call for sourceChain2
+	destExtended.EXPECT().ExtendedGetLatestValue(
+		mock.Anything,
+		consts.ContractNameOffRamp,
+		consts.MethodNameGetSourceChainConfig,
+		primitives.Unconfirmed,
+		map[string]any{"sourceChainSelector": sourceChain2},
+		mock.Anything,
+	).Return(nil).Run(func(
+		ctx context.Context,
+		contractName,
+		methodName string,
+		confidenceLevel primitives.ConfidenceLevel,
+		params,
+		returnVal interface{}) {
+		v := returnVal.(*sourceChainConfig)
+		v.OnRamp = s2Onramp
+		v.IsEnabled = true
+	})
+	// mock the call to get the nonce manager
+	destExtended.EXPECT().ExtendedGetLatestValue(
+		mock.Anything,
+		consts.ContractNameOffRamp,
+		consts.MethodNameOfframpGetStaticConfig,
+		primitives.Unconfirmed,
+		map[string]any{},
+		mock.Anything,
+	).Return(nil).Run(func(
+		ctx context.Context,
+		contractName,
+		methodName string,
+		confidenceLevel primitives.ConfidenceLevel,
+		params,
+		returnVal interface{}) {
+		v := returnVal.(*offrampStaticChainConfig)
+		v.NonceManager = destNonceMgr
+	})
+	defer destExtended.AssertExpectations(t)
```

Comment:
Is this call really needed? I think tooling calls it at the end of the test anyway, doesn't it?

Reply:
Seems like you're right, this is called in the `Cleanup` function of each mock, will remove.